src/app/features/movie/movie.container.ts

```diff
@@ -0,0 +1,40 @@
+import { Component } from '@angular/core';
+import { UsersService } from '../../core/services/users.service';
+import { ActivatedRoute } from '@angular/router';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { AsyncPipe } from '@angular/common';
+import { Observable } from 'rxjs';
+import { MoviesService } from '../../core/services/movies.service';
+import { Movie, UserMovie } from '../../core/models/movie.interface';
+import { MovieComponent } from './movie.component';
+
+@Component({
+  standalone: true,
+  imports: [AsyncPipe, MovieComponent],
+  template: `
+    <app-movie
+      [id]="movieId"
+      [movieData]="(movieData$ | async) ?? undefined"
+      [userMovieData]="(userMovieData$ | async) ?? undefined" />
+  `,
+})
+export class MovieContainerComponent {
+  movieId!: string;
+  movieData$?: Observable<Movie | undefined>;
+  userMovieData$?: Observable<UserMovie | undefined>;
+
+  constructor(
+    private moviesService: MoviesService,
+    private usersService: UsersService,
+    private route: ActivatedRoute,
+  ) {
+    this.route.paramMap.pipe(takeUntilDestroyed()).subscribe((params) => {
+      const paramId = params.get('movieId');
+      if (paramId) {
+        this.movieId = paramId;
+        this.movieData$ = this.moviesService.get(this.movieId);
+        this.userMovieData$ = this.usersService.getUserMovie(this.movieId);
+      }
+    });
+  }
```

Comment:
It feels super wrong to use the subscribe to override observables. You ignore the streaming nature of rxjs. api driven I would prefer something like the following as a beginning

In pseudo code:

```
  this.movieData$ = this.route.paramMap.pipe(switchMap(paramid => this.moviesService.get(paramid))
  this.userMovieData$ =this.route.paramMap.pipe(switchMap(paramid => this.moviesService.get(paramdId))
```

you don't even need a take until destroyed because async will handle unsubscribing from the sources


also there is a common pattern: do not use the subscribe but use a tap for side effects.

```
something.subscribe(value => console.log(value))
// better
something.pipe(
  tap(value => console.log(value)
).subscribe()
)
```

That way you can easily defer the subscription or combine it with other streams

```
strean1$ =  something.pipe(
  tap(value => console.log(value)
)

strean1$.pipe(whatever).subscribe()

```